lib/api/controllers/remoteActions/managePlugins.js

```diff
@@ -0,0 +1,591 @@
+       between different plugins
+       */
+      return db.updateMapping(
+        kuzzleConfiguration.pluginsManager.dataCollection,
+        {
+          properties: {
+            config: {
+              enabled: false
+            }
+          }
+        }
+      );
+    })
+    .catch(err => {
+      // ignoring error if it's raised because the index already exists
+      if (err.status === 400) {
+        return q();
+      }
+
+      return q.reject(err);
+    });
+}
+
+/**
+ * Returns the list of plugins to install
+ * The returned list is in the following format:
+ *   [
+ *     "plugin-name": {
+ *       "how to": "get this plugin"
+ *     }
+ *   ]
+ *
+ * @param {Object} db - database service
+ * @param {Object} cfg - Kuzzle configuration
+ * @returns Promise
+ */
+function getPluginsList(db, cfg) {
+  return db
+    .search(cfg.pluginsManager.dataCollection)
+    .then(result => {
+      var plugins = {};
+
+      if (result.total === 0) {
+        return cfg.pluginsManager.defaultPlugins;
+      }
+
+      result.hits.forEach(p => {
+        plugins[p._id] = p._source;
+      });
+
+      return plugins;
+    });
+}
+
+/**
+ * Install plugins and load their configuration into the database
+ *
+ * @param plugin to install. May be undefined (install all listed plugins)
+ * @param options - arguments supplied by the user on the command-line
+ * @param dbService - handler to Kuzzle's database
+ * @param kuzzleConfiguration - Kuzzle configuratioin
+ * @returns {Promise}
+ */
+function installPlugins(plugin, options, dbService, kuzzleConfiguration) {
+  var
+    p = {},
+    pluginsListPromise,
+    pluginsList;
+
+  if (plugin) {
+    p[plugin] = {};
+    ['npmVersion', 'gitUrl', 'path'].forEach(tag => {
+      if (options[tag] !== undefined) {
+        p[plugin][tag] = options[tag];
+      }
+    });
+
+    p[plugin].activated = options.activated;
+
+    pluginsListPromise = q(p);
+  }
+  else {
+    pluginsListPromise = getPluginsList(dbService, kuzzleConfiguration);
+  }
+
+  return pluginsListPromise
+    .then(plugins => {
+      pluginsList = plugins;
+      return acquirePlugins(plugins);
+    })
+    .then(() => {
+      console.log('███ kuzzle-plugins: Updating plugins configuration...');
+
+      return updatePluginsConfiguration(
+        dbService,
+        kuzzleConfiguration.pluginsManager.dataCollection,
+        pluginsList);
+    });
+}
+
+/**
+ * Import a configuration from a given file for a given plugin
+ * @param db
+ * @param plugin
+ * @param filePath
+ */
+function importPluginConfiguration(db, plugin, filePath, kuzzleConfiguration) {
+  var
+    content;
+
+  try {
+    content = fs.readFileSync(filePath, 'UTF-8');
+  } catch (err) {
+    return q.reject(new Error('File not found: ' + filePath));
+  }
+
+  try {
+    content = JSON.parse(content);
+  }
+  catch (err) {
+    return q.reject(new Error('Unable to parse ' + content + '. Expected: JSON Object'));
```

Comment:
If the file content is large enough, returning it may make life difficult for administrators trying to guess why parsing it failed.
I think returning the file name and the error message would be better.